app/managers/websocket_manager.ts

```diff
@@ -45,44 +47,49 @@ class WebsocketManager {
             ({serverUrl, token}) => {
                 try {
                     DatabaseManager.getServerDatabaseAndOperator(serverUrl);
-                    this.createClient(serverUrl, token, 0);
+                    this.createClient(serverUrl, token);
                 } catch (error) {
                     logError('WebsocketManager init error', error);
                 }
             },
         );
 
-        AppState.addEventListener('change', this.onAppStateChange);
-        NetInfo.addEventListener(this.onNetStateChange);
+        this.appStateSubscription?.remove();
+        this.netStateSubscription?.();
+
+        this.appStateSubscription = AppState.addEventListener('change', this.onAppStateChange);
+        this.netStateSubscription = NetInfo.addEventListener(this.onNetStateChange);
     };
 
     public invalidateClient = (serverUrl: string) => {
-        this.clients[serverUrl]?.close();
+        this.clients[serverUrl]?.close(true);
         this.clients[serverUrl]?.invalidate();
-        if (this.connectionTimerIDs[serverUrl]) {
-            this.connectionTimerIDs[serverUrl].cancel();
-        }
+        clearTimeout(this.connectionTimerIDs[serverUrl]);
         delete this.clients[serverUrl];
         delete this.firstConnectionSynced[serverUrl];
 
+        // We don't remove the connected subject so any potential client invalidation
+        // and subsequent creation of the client can still be observed by the component.
+        // Being purist, this is a memory leak, since we never clean any server url,
+        // but since this information lives in memory and we don't expect many servers
+        // to be added and removed in one single session, this should be fine.
         this.getConnectedSubject(serverUrl).next('not_connected');
```

Comment:
This is the reason why we may see the "not connected" banner while still receiving everything alright.